src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/PlotterControls.java

```diff
@@ -1,90 +1,151 @@
+	private ButtonIcon bEmergencyStop;
+	private final JProgressBar progress = new JProgressBar(0, 100);
 
 	private boolean isRunning = false;
 	private boolean penIsUpBeforePause = false;
 
-	public PlotterControls(Plotter plotter, Turtle turtle) {
+	public PlotterControls(Plotter plotter, Turtle turtle, Window parentWindow) {
 		super();
 		myPlotter = plotter;
 		myTurtle = turtle;
 
 		jogInterface = new JogInterface(plotter);
-		marlinInterface = new MarlinPlotterInterface(plotter);
+		marlinInterface = new MarlinPlotterInterface(plotter, connectionButton);
 		programInterface = new ProgramInterface(plotter, turtle);
 
-		JTabbedPane pane = new JTabbedPane();
-		pane.addTab(JogInterface.class.getSimpleName(), jogInterface);
-		pane.addTab(MarlinInterface.class.getSimpleName(), marlinInterface);
-		pane.addTab(ProgramInterface.class.getSimpleName(), programInterface);
+		JTabbedPane tabbedPane = new JTabbedPane();
+		jogInterface.setPreferredSize(new Dimension(580, 300));
+		tabbedPane.addTab(Translator.get("PlotterControls.JogTab"), jogInterface);
+		tabbedPane.addTab(Translator.get("PlotterControls.MarlinTab"), marlinInterface);
+		tabbedPane.addTab(Translator.get("PlotterControls.ProgramTab"), programInterface);
+
+		CollapsiblePanel collapsiblePanel = new CollapsiblePanel(parentWindow, Translator.get("PlotterControls.AdvancedControls"), 570);
+		collapsiblePanel.add(tabbedPane);
 
 		this.setLayout(new BorderLayout());
-		this.add(pane, BorderLayout.CENTER);
-		this.add(getToolBar(), BorderLayout.NORTH);
+		this.add(collapsiblePanel, BorderLayout.CENTER);
+		this.add(getButtonsPanels(), BorderLayout.NORTH);
 		this.add(progress, BorderLayout.SOUTH);
 
-		marlinInterface.addListener((e) -> {
-			if (e.getActionCommand().contentEquals(MarlinInterface.IDLE)) {
-				// logger.debug("PlotterControls heard idle");
-				if (isRunning) {
-					// logger.debug("PlotterControls is running");
-					step();
-				}
+		marlinInterface.addListener(e -> {
+			if (e.getActionCommand().equals(MarlinInterface.IDLE) && isRunning) {
+				step();
+			} else if (e.getActionCommand().equals(MarlinInterface.ERROR)) {
+				JOptionPane.showMessageDialog(this,  Translator.get("PlotterControls.FatalError"), Translator.get("PlotterControls.FatalErrorTitle"), JOptionPane.ERROR_MESSAGE);
 			}
 			updateProgressBar();
 		});
 	}
 
-	private JToolBar getToolBar() {
-		JToolBar bar = new JToolBar();
-		bar.setFloatable(false);
-		bar.add(bFindHome);
-		bar.addSeparator();
-		bar.add(bRewind);
-		bar.add(bStart);
-		bar.add(bPause);
-		bar.add(bStep);
+	private JPanel getButtonsPanels() {
+		JPanel panel = new JPanel();
+		panel.setLayout(new BorderLayout());
+		panel.add(getConnectPanel(), BorderLayout.NORTH);
+		panel.add(getDrawPanel(), BorderLayout.CENTER);
+		return panel;
+	}
+
+	private JPanel getConnectPanel() {
+		JPanel panel = new JPanel();
+		Border border = BorderFactory.createTitledBorder(Translator.get("PlotterControls.ConnectControls"));
```

Comment:
I don't like taking two classes and smashing them into one.  PlotterControls should delegate these responsibilities to the ChooseConnection.  

Having said that, I like how you improved the look and feel.  It's much better without a popup.

Reply:
You mean having a new panel `ChooseConnectionPanel` which take care of all of this ?

Reply:
Yeah.  there used to be a ChooseConnection.  half of it became ConnectionButtonand half of it became ... like, what is connectionComboBox here for if there's also `ConnectionButton.connectionComboBox`?

Reply:
You're right, renaming `ConnectionButton` to `ChooseConnection` and your comments show me that `ChooseConnection` is not just a button but a panel.
I moved the `combobox` into `ChooseConnection`. It's better now ! Thanks